app/submitter/convert_payload_0_0_1.py

```diff
@@ -18,10 +19,9 @@ def convert_answers_to_payload_0_0_1(answer_store, schema, routing_path):
     for location in routing_path:
         answer_ids = schema.get_answer_ids_for_block(location.block_id)
         answers_in_block = answer_store.filter(answer_ids)
-        answer_schema_list = schema.get_answers_by_id_for_block(location.block_id)
 
         for answer in answers_in_block:
-            answer_schema = answer_schema_list[answer['answer_id']]
+            answer_schema = schema.get_answer_with_context(answer['answer_id'], metadata, answer_store)
```

Comment:
I think we should look to remove the `get_answer_with_context` method from questionnaire schema as it's introducing a dependency on metadata and answer_store. The `get_answer_with_context` method uses the answer id to get the block id, then gets the block, and evaluates that to determine which question was displayed. Instead we could get the block to get the block and call `choose_question_to_display` here to determine the question, and from that the answers.